### ISSUE TITLE
[Feature Request] Port to non-Linux platforms

### ISSUE THREAD

**Issue by rodneyrod** [pdf/kdeconnect-chrome-extension#4]

**Comment by pdf** [comment]
Any in particular?

**Comment by rodneyrod** [comment]
Windows and OSX would probably be the best other platforms to target, considering they have no KDE Connect support as it currently stands.

**Comment by pdf** [comment]
This project uses the dbus interface to communicate with a running KDE Connect instance, so porting to platforms that don't have a KDE Connect implementation is not possible.

**Comment by rodneyrod** [comment]
Right, issue is upstream then. I'll close this until it is resolved there.

**Comment by rodneyrod** [comment]
KDE Connect is at least partly ported to Windows upstream, so opening this issue back up to track this.

https://binary-factory.kde.org/job/kdeconnect-kde_Nightly_win32/

**Comment by pdf** [comment]
Can you confirm whether DBus is also ported, and whether this KDE Connect port uses it?

If DBus is also working on Windows, it's should be a simple matter of compiling this for Windows, and adding some install paths to the installer for various browsers.  If DBus is not working on Windows, nothing can/will be done here to support it.

**Comment by rodneyrod** [comment]
@pdf I can confirm that KDE Connect for Windows does use DBus, or rather a port of it.
Should be running as dbus-daemon.exe

**Comment by pdf** [comment]
I spent a bit of time looking at this, and it seems like it ought to be doable. I have the installation with reg keys etc sorted out, but not connecting to DBus on Windows - it apparently only works over TCP using a noncefile for auth, and I'm not sure how to connect to the session that KDE Connect creates.  This is quite different to other platforms.

**Comment by shamoi** [comment]
Any update on on this, anyone ?

**Comment by firefinchdev** [comment]
KDE Connect now has a macos version which uses dbus interface. Although there is not a stable release yet, but the builds are working perfectly. Please port this extension to macos. Would be really great. Thank you.

**Comment by Cerberus-tm** [comment]
Not sure if this is appropriate, but I'd absolutely love to be able to use KDE Connect from Firefox on Windows. The KDE Connect desktop application works on Windows (though it is not always reliable).

**Comment by kernelb00t** [comment]
You can check the [Bitwarden client sources](https://github.com/bitwarden/clients), the extension talks with the host application to spawn a Windows Hello dialog. Implementing Windows/MacOS will very likely use a separate backend to talk to the KDE Connect host application.
